### PR TITLE
[bit7z] new port bit7z

### PR DESCRIPTION
**Pull request by uuiid**
https://github.com/rikyoz/bit7z

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

run `./vcpkg x-add-version --all ` crashes.

```
error: bit7z is not properly formatted
Run `vcpkg format-manifest ports/bit7z/vcpkg.json` to format the file
Don't forget to commit the result!

internal error: D:\a\_work\1\s\src\vcpkg\commands.add-version.cpp(437): vcpkg has crashed; no additional details are available.
Please open an issue at https://github.com/microsoft/vcpkg/issues/new?template=other-type-of-bug-report.md&labels=category:vcpkg-bug with detailed steps to reproduce the problem.
```

### CODE REVIEW

**Comment by uuiid** [comment]
@microsoft-github-policy-service agree

**Comment by dg0yt** [comment]
> error: bit7z is not properly formatted
> Run `vcpkg format-manifest ports/bit7z/vcpkg.json` to format the file

DId you try that?

File: ports/bit7z/0001-Fix-using-vcpkg-7z-package-and-install.patch [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
++    ARCHIVE DESTINATION lib
++    LIBRARY DESTINATION lib
++    RUNTIME DESTINATION bin
++    FILE_SET bit7z_headers
++)
++
++
++include(CMakePackageConfigHelpers)
++configure_package_config_file(${PROJECT_NAME}-config.cmake.in ${PROJECT_NAME}-config.cmake INSTALL_DESTINATION share/${PROJECT_NAME})
++install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake DESTINATION share/${PROJECT_NAME})
```
Comment (dg0yt) [inline ports/bit7z/0001-Fix-using-vcpkg-7z-package-and-install.patch:75]: If you patch-in cmake config export, package name and targets must be in unofficial namespaces: 
https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#add-cmake-exports-in-an-unofficial--namespace

File: ports/bit7z/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    "link-libcpp": {
+      "description": "Enable linking to libc++",
+      "supports": "!windows"
+    },
+    "auto-prefix-long-paths": {
+      "description": "Enable automatically prepend paths with Windows long path prefix",
+      "supports": "windows"
+    }
+  }
+}
```
Comment (jimwang118) [inline ports/bit7z/vcpkg.json:52]: Please add a blank line at the end of the file.

File: ports/bit7z/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{
+  "name": "bit7z",
+  "version": "4.0.0-rc",
+  "port-version": 1,
```
Comment (jimwang118) [inline ports/bit7z/vcpkg.json:4]:
```suggestion
```

File: ports/bit7z/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
+{
+  "name": "bit7z",
+  "version": "4.0.0-rc",
```
Comment (jimwang118) [inline ports/bit7z/vcpkg.json:3]: 4.0.0 is a beta version, should we use the stable release version?

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

**Comment by jimwang118** [comment]
@uuiid Thanks for your PR, Is work still being done for this PR?

**Comment by dan-shaw** [comment]
Closing this PR for now, feel free to open a new PR if you are still working on it

**Pull request closed without merging**